chapter07/chapter07.adoc

```diff
@@ -0,0 +1,276 @@
+
+4. *Timer*: It is a specialized metric type that aggregates timing durations and provides count, total time, mean, minimum, and maximum duration, and it usually reports the duration distribution as well. It effectively combines a Histogram and a Meter, measuring the rate of events and the time duration of each event. Timers are invaluable for tracking the duration of certain activities or operations within your application, such as processing time or method execution time.
+
+By leveraging these metrics, developers and operators can gain a deeper understanding of how their microservices are performing. They can use this information to identify areas where improvements can be made and optimize their microservices' performance. 
+
+=== MicroProfile Metrics Dependency
+
+If you're using Maven, add the following dependency to your pom.xml file located in the root folder of your project:
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.eclipse.microprofile.metrics</groupId>
+  <artifactId>microprofile-metrics-api</artifactId>
+  <version>5.1.1</version>
+</dependency>
+----
+
+For Gradle, add the corresponding dependency to your build.gradle file located within the root folder of your project:
+
+[source]
+----
+dependencies {
+    providedCompile 'org.eclipse.microprofile.metrics:microprofile-metrics-api:5.1.1'
+}
+----
+  
+=== Metrics Annotations
+
+MicroProfile Metrics defines a set of annotations to be used for exposing metrics. These annotations can be used on classes, methods, or fields. Table 7-1 shows the list of Metrics Annotation along with their descriptions.
+
+[options="header", cols="2,5"]
+|===
+| Annotation | Description
+
+| `@Timed`
+| It times how long a method takes to execute and exposes this information as a metric.
+
+| `@Counted`
+| It tracks how many times a method is invoked and exposes this information as a metric.
+
+| `@Gauge`
+| It allows you to expose a custom metric that can be any value. It is useful for exposing application-specific metrics.
+|===
+
+Besides annotations, MicroProfile Metrics also defines a set of programmatic APIs for working with metrics. These APIs can be used to register custom metrics or access existing metrics.
+
+=== Using Metadata for Metrics
+
+In MicroProfile Metrics, metadata plays a crucial role in defining, understanding, and managing the various metrics that applications expose. Metadata essentially provides descriptive information about a metric, such as its name, description, type, unit of measurement, and additional tags or labels. This information not only aids in the identification and categorization of metrics but also enhances their interpretability, making it easier for developers, operators, and monitoring tools to understand what each metric represents and how it should be used.
```

Comment:
Strictly. speaking the `Metadata` object doesn't actually hold tag/label data. That belongs under the `MetricID` object. Both work together to work as _metadata_ I suppose. The rest of the guide/tutorial doesn't seem to get into the nitty gritty details of using the API so this we can _ignore_ I suppose. wdyt @donbourne

Reply:
As advised, removing this section about Using Metadata for Metrics.